### PR TITLE
Implement email input validation and styling

### PR DESCRIPTION
**Pull request by hannajones**
This PR adds a custom error message tooltip to the email input in the sign in step. The styling is largely taken from the FxA sign in form, which I believe was the inspiration for the mockups.

Screenshot 2023-05-18 at 10 42 47 AM

### CODE REVIEW

File: kitsune/sumo/static/sumo/js/form-wizard-sign-in-step.js [hunk 1]
```diff
@@ -61,16 +68,61 @@ export class SignInStep extends BaseFormStep {
...
+          this.handleShowError();
+          event.preventDefault();
+        }
+        break;
+      }
+    }
+  }
+
+  handleShowError() {
+    this.#emailErrorEl.textContent = gettext("Valid email required");
```
[the remark this answers is not in the document]
Comment (mikeconley) [reply kitsune/sumo/static/sumo/js/form-wizard-sign-in-step.js:113]: I think we might just need to run the copy by Fabi and/or sfrench in https://docs.google.com/document/d/1bVcQz5tDDfbPf9t6zXfgjmHfkNqXfWW_xBqw4WV5TTw/edit

File: kitsune/sumo/static/sumo/scss/form-step.styles.scss [hunk 2]
```diff
@@ -16,3 +18,42 @@
...
+  color: var(--color-white);
+  padding: 8px;
+  position: absolute;
+  top: -34px;
+}
+
+.error-tooltip::before {
+  background-color: p.$color-red-60;
+  bottom: -8px;
+  content: ".";
```
Comment (mikeconley) [inline kitsune/sumo/static/sumo/scss/form-step.styles.scss:51]: Does this actually need a character here? I think `content: ""` should do - although if this is being cribbed from the Protocol styles, maybe best to just keep what they have.

File: kitsune/sumo/static/sumo/scss/form-step.styles.scss [hunk 3]
```diff
@@ -16,3 +18,42 @@
   line-height: 1.14;
   color: var(--color-text-form-step);
 }
+
+.hidden {
+  visibility: hidden;
+}
+
+.error-tooltip:not(.hidden) + input:invalid,
```
[the remark this answers is not in the document]
Comment (mikeconley) [reply kitsune/sumo/static/sumo/scss/form-step.styles.scss:26]: What tooltip do we need for the setup device step?

**Comment by mikeconley** [comment]
It also looks like this will close mozilla/sumo#1300.

File: kitsune/sumo/static/sumo/scss/form-step.styles.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (hannajones) [reply kitsune/sumo/static/sumo/scss/form-step.styles.scss:26]: hmm that first link didn't work, here's a screenshot:

Screenshot 2023-05-18 at 11 20 02 AM

File: kitsune/sumo/static/sumo/js/form-wizard-sign-in-step.js [hunk 4]
```diff
@@ -61,16 +68,61 @@ export class SignInStep extends BaseFormStep {
...
+      case "blur": {
+        if (!this.#emailEl.validity.valid) {
+          this.handleShowError();
+        }
+        break;
+      }
+      case "input": {
+        if (this.#emailEl.value?.trim()) {
+          this.#emailErrorEl.classList.add("hidden");
+          this.#emailErrorEl.textContent = "";
```
[the remark this answers is not in the document]
Comment (hannajones) [reply kitsune/sumo/static/sumo/js/form-wizard-sign-in-step.js:98]: That's a good point - I don't think we're planning on changing the messages, so I'll just go ahead and use this as the default/only text for the tooltip.

File: kitsune/sumo/static/sumo/scss/form-step.styles.scss (hunk 2 above)
Comment (hannajones) [reply kitsune/sumo/static/sumo/scss/form-step.styles.scss:51]: Good catch! This was largely cribbed from FxA styles, but I've already been tweaking it a bit to fit our case.

File: kitsune/sumo/static/sumo/js/form-wizard-sign-in-step.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (escattone) [reply kitsune/sumo/static/sumo/js/form-wizard-sign-in-step.js]: @hannajones @mikeconley I think within the next few days, we (the SUMO engineering team) should do an extraction of localizable strings from all of our JS, Python, Jinja templates, and Svelte code and submit the new strings to Pontoon. We do that routinely, but we should probably do that soon for this project so Pontoon localizers have time to work on the translations.

**Pull request merged**